central/authprovider/datastore/datastore_impl.go

```diff
@@ -25,13 +25,13 @@ type datastoreImpl struct {
 	storage store.Store
 }
 
-// GetAllAuthProviders retrieves authProviders.
-func (b *datastoreImpl) GetAllAuthProviders(ctx context.Context) ([]*storage.AuthProvider, error) {
+// GetAllAuthProviders retrieves authProviders and process each one with provided function.
+func (b *datastoreImpl) GetAllAuthProviders(ctx context.Context, fn func(obj *storage.AuthProvider) error) error {
```

Comment:
This function does not return an iterator because of other existing similar implementations?

Reply:
I don't know how to handle db errors here. If accept we ignore them we can use iterator. If not we could use this approach https://pkg.go.dev/github.com/bobg/seqs#Lines but it has it's own issues. Since I don't know how to handle errors here I decided to do a callback.
cc: @mtodor

Reply:
Well, errors could be collected separately, which makes using an iterator less elegant.